### PR TITLE
Truncations v1.0

### PR DESCRIPTION
**Pull request by mahrud**
This is a significant rewrite of the Truncations package. The main improvements are:

- reorganizing and clarifying the algorithms with comments
- adding support for Cox rings of simplicial normal toric varieties
- fixing truncation of modules with relations
- making `truncate` accept arbitrary options (this involves minor changes to some other packages)
- significantly optimizing a whole bunch of algorithms
- adding a bunch of new tests for things that were wrong before
- fixing the documentation to correctly describe what truncation does
- also closing #1945.

Small note: commit aec5e02 is a short-term patch for a much larger problem that should be resolved in https://github.com/Macaulay2/M2/issues/1358.

cc: @mikestillman, @eisenbud

### CODE REVIEW

**Comment by mahrud** [comment]
The only failing test is from SpecialFanoFourfolds. @DanGrayson you might want to take a look at this also, while Mike reviews it, as there are some changes in the Core included here (mainly to fix a few documentation issues).

**Comment by DanGrayson** [comment]
> @DanGrayson you might want to take a look at this also

will do ...

File: M2/Macaulay2/m2/lists.m2 [hunk 1]
```diff
@@ -175,6 +175,8 @@ rotate(ZZ,VisibleList) := (n,s) -> (
 sort List :=  opts -> internalsort
 rsort List := opts -> internalrsort
 
+List << List := (A, B) -> all(min(#A, #B), i -> A#i <= B#i)
```
Comment (DanGrayson) [inline M2/Macaulay2/m2/lists.m2:178]: Let's not add this.  We already have comparison for lists with the operator `<=`, and `<<` doesn't look like a comparison operator.

**Comment by DanGrayson** [comment]
> > @DanGrayson you might want to take a look at this also
> 
> will do ...

I made one comment.

File: M2/Macaulay2/m2/lists.m2 (hunk 1 above)
Comment (mahrud) [reply M2/Macaulay2/m2/lists.m2:178]: This was discussed in https://github.com/Macaulay2/M2/issues/1945. Quoting your comment there:
> I can't think of a better use for List << List. In the documentation, please compare it to the behavior of List ? List and friends (such as List <= List).

Comment (DanGrayson) [reply M2/Macaulay2/m2/lists.m2:178]: Oh, okay, never mind.

**Comment by mikestillman** [comment]
@DanGrayson can you resolve the conflict in changes.m2?  Then accept this pull request?

**Pull request merged**